### PR TITLE
Add support for MultiAddress in ISA (selected inputs only)

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/client/api/block_builder/input_selection/requirement/sender.rs [hunk 1]
```diff
@@ -42,6 +42,30 @@ impl InputSelection {
...
+                            .output
+                            .required_and_unlocked_address(self.slot_index, input.output_id())?
+                            .0;
+
+                        if &required_address == weight_address.address() {
+                            cumulative_weight += weight_address.weight() as u16;
+                            break;
+                        }
+                    }
+                }
```
Comment (Alex6323) [inline sdk/src/client/api/block_builder/input_selection/requirement/sender.rs]: could be done more efficiently I think (but depends on how costly the fn call and how likely the worst case scenario is)

Comment (thibault-martinez) [reply sdk/src/client/api/block_builder/input_selection/requirement/sender.rs]: How? Without a substantial change to ISA I'm not sure. Also this is a temporary feature as suggested by the TODO.

Comment (Alex6323) [reply sdk/src/client/api/block_builder/input_selection/requirement/sender.rs]: I would probably also check the threshold in the loop so we can break if the threshold is surpassed already, unless we really need the exact amount of cumulative weight.

Comment (thibault-martinez) [reply sdk/src/client/api/block_builder/input_selection/requirement/sender.rs]: Will do

Comment (thibault-martinez) [reply sdk/src/client/api/block_builder/input_selection/requirement/sender.rs]: Done

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**